File: Data Structures & Algorithms/top-k-elements-in-list/submission-0.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        hashmap = {}
        for i in nums:
            hashmap[i]=hashmap.get(i,0)+1
        list=sorted(hashmap.keys(),key=lambda x:hashmap[x],reverse=True)
        return list[:k]

        

File: Data Structures & Algorithms/find-duplicate-integer/submission-0.java
class Solution {
    public int findDuplicate(int[] nums) 
    {
        HashMap<Integer,Integer> map = new HashMap<>();
        for(int i:nums)
        {
            map.put(i,map.getOrDefault(i,0)+1);
        }
        int maxkey = Integer.MIN_VALUE;
        int maxval = Integer.MIN_VALUE;
        for(Map.Entry<Integer,Integer>entry:map.entrySet())
        {
            if(entry.getValue()>maxval)
            {
                maxval=entry.getValue();
                maxkey=entry.getKey();
            }
        }
        return maxkey;
    }
}
